### PR TITLE
Add `Retrofit` & `Flow`

### PR DESCRIPTION
**Pull request by alesrazym**

### CODE REVIEW

File: app/src/main/java/cz/quanti/razym/rocketapp/system/RocketListFragment.kt [hunk 1]
```diff
@@ -23,8 +29,30 @@ class RocketListFragment : Fragment() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        viewModel.rocketLiveData.observe(this) { rockets ->
-            binding.rocketList.adapter = RocketListAdapter(rockets)
+        lifecycleScope.launch {
```
Comment (havlisimo) [inline app/src/main/java/cz/quanti/razym/rocketapp/system/RocketListFragment.kt:32]: podle me tohle muze zpusobovat problemy a bylo by lepsi pouzit
viewLifecycleOwner.lifecycleScope.launch {  } v onViewCreated. Konkretne se muze stat ze binding bude null a appka crashne.

File: app/src/main/java/cz/quanti/razym/rocketapp/presentation/RocketListViewModel.kt [hunk 2]
```diff
@@ -1,13 +1,59 @@
...
+    init {
+        fetchRockets()
+    }
+
+    fun fetchRockets() {
+        viewModelScope.launch {
+            repo.getRockets()
+                .asResult()
+                .collect { result ->
+                    val genreUiState = when (result) {
```
Comment (havlisimo) [inline app/src/main/java/cz/quanti/razym/rocketapp/presentation/RocketListViewModel.kt]: co znamena genre?

**Review by havlisimo — CHANGES_REQUESTED** [review]

File: app/src/main/java/cz/quanti/razym/rocketapp/presentation/RocketListViewModel.kt (hunk 2 above)
Comment (alesrazym) [reply app/src/main/java/cz/quanti/razym/rocketapp/presentation/RocketListViewModel.kt]: pozustatek ze samplu :/

File: app/src/main/java/cz/quanti/razym/rocketapp/system/RocketListFragment.kt (hunk 1 above)
Comment (alesrazym) [reply app/src/main/java/cz/quanti/razym/rocketapp/system/RocketListFragment.kt:32]: Zkontroloval jsem dokumentaci, bude to v poradku. `repeatOnLifecycle` se zrusi pri prechodu do STOP stavu
> repeatOnLifecycle launches the block in a new coroutine every time the
lifecycle is in the STARTED state (or above) and cancels it when it's STOPPED.

https://developer.android.com/topic/libraries/architecture/coroutines#restart

Jeste je mozne vyuzit o jednu uroven zanoreni mene s Flow.flowWithLifecycle() https://developer.android.com/reference/kotlin/androidx/lifecycle/package-summary#flowwithlifecycle, tak to vyuziju a trochu rfc.

Comment (alesrazym) [reply app/src/main/java/cz/quanti/razym/rocketapp/system/RocketListFragment.kt:32]: Lze vyuzit `lifecycle.whenXxx`, nize v clanku https://developer.android.com/topic/libraries/architecture/coroutines#suspend maji pouziti pro ochranu pristupu, ale pouziti whenXxx nedoporucujou.

**Review by havlisimo — APPROVED** [review]

**Pull request merged**